modules/broker-coordination/src/main/java/org/wso2/broker/coordination/rdbms/RdbmsCoordinationStrategy.java

```diff
@@ -108,23 +109,38 @@
      */
     private final ScheduledExecutorService scheduledExecutorService;
 
+    /**
+     * Listeners expecting notification on coordinator change.
+     */
+    private List<RdbmsCoordinationListener> coordinationListeners = new ArrayList<>();
+
     /**
      * Default constructor.
      *
      * @param rdbmsCoordinationDaoImpl       the RdbmsCoordinationDaoImpl to use for communication with the database
-     * @param rdbmsCoordinationConfiguration the configuration for RDBMS coordination
+     * @param rdbmsCoordinationOptions       the configuration for RDBMS coordination
      */
     public RdbmsCoordinationStrategy(RdbmsCoordinationDaoImpl rdbmsCoordinationDaoImpl,
-                                     CoordinationConfiguration.RdbmsCoordinationConfiguration
-                                             rdbmsCoordinationConfiguration) {
+                                     Map<String, String> rdbmsCoordinationOptions) {
         ThreadFactory namedThreadFactory = new ThreadFactoryBuilder().setNameFormat("RdbmsCoordinationStrategy-%d")
                 .build();
         threadExecutor = Executors.newSingleThreadExecutor(namedThreadFactory);
         scheduledExecutorService = Executors.newSingleThreadScheduledExecutor();
 
-        heartBeatInterval = rdbmsCoordinationConfiguration.getHeartbeatInterval();
-        coordinatorEntryCreationWaitTime = rdbmsCoordinationConfiguration.getCoordinatorEntryCreationWaitTime();
-        //(heartBeatInterval / 2) + 1;
+        if (rdbmsCoordinationOptions != null) {
```

Comment:
Can rdbmsCoordinationOptions be null? Can't we return an empty map where we return null so that we can avoid this null check?